tests/common.go

```
@@ -1693,33 +1693,43 @@ func ValidatePureVolumeLargeNumOfClones(ctx *scheduler.Context, errChan ...*chan
+		var vols []*volume.Volume
+		var err error
+		Step(fmt.Sprintf("get %s app's pure volumes", ctx.App.Key), func() {
+			vols, err = Inst().S.GetPureVolumes(ctx, "pure_block")
 			processError(err, errChan...)
-		}
+		})
+		if len(vols) == 0 {
+			log.Warnf("No FlashArray DirectAccess volumes, skipping")
+		} else {
 
-		if len(pools) == 0 {
-			err = fmt.Errorf("length of pools should be greater than 0")
-			processError(err, errChan...)
-		}
-		for _, pool := range pools {
-			initialPoolSize := pool.TotalSize / units.GiB
-			err = Inst().V.ResizeStoragePoolByPercentage(pool.Uuid, opsapi.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK, 20)
+			pools, err := Inst().V.ListStoragePools(metav1.LabelSelector{})
 			if err != nil {
-				err = fmt.Errorf("error initiating pool [%v ] %v: [%v]", pool.Uuid, opsapi.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK, err.Error())
+				err = fmt.Errorf("error getting storage pools list. Err: %v", err)
 				log.Error(err.Error())
-			} else {
-				err = waitForPoolToBeResized(initialPoolSize, pool.Uuid)
+				processError(err, errChan...)
+			}
+
+			if len(pools) == 0 {
+				err = fmt.Errorf("length of pools should be greater than 0")
+				processError(err, errChan...)
+			}
+			for _, pool := range pools {
+				initialPoolSize := pool.TotalSize / units.GiB
+				err = Inst().V.ResizeStoragePoolByPercentage(pool.Uuid, opsapi.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK, 20)
 				if err != nil {
-					err = fmt.Errorf("pool [%v] %v failed. Error: %v", pool.Uuid, opsapi.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK, err)
-					log.Error(err)
+					err = fmt.Errorf("error initiating pool [%v ] %v: [%v]", pool.Uuid, opsapi.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK, err.Error())
+					log.Error(err.Error())
+				} else {
+					err = waitForPoolToBeResized(initialPoolSize, pool.Uuid)
+					if err != nil {
+						err = fmt.Errorf("pool [%v] %v failed. Error: %v", pool.Uuid, opsapi.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK, err)
+						log.Error(err)
```

can we fail instead of log.Error?

---

This is old piece of code not changed by this PR. This URL shows exact changes done as part of current PR: https://github.com/portworx/torpedo/pull/2456/files?diff=split&w=1